R/gbif_citation.R

```diff
@@ -77,23 +44,22 @@ gbif_citation <- function(x) {
 
 #' @export
 gbif_citation.gbif <- function(x) {
-  if (!inherits(x, "data.frame")) {
-    x <- x$data
-  }
-  dkeys <- unique(suppressWarnings(x$datasetKey))
-  if (is.null(dkeys)) {
-    occ_keys <- x$key
-    if (!is.null(occ_keys)) {
-      dkeys <- unique(vapply(occ_get(as.numeric(unique(occ_keys)),
-        fields = "all"), "[[", "", c('data', 'datasetKey')))
-    } else {
-      stop("No 'datasetKey' or 'key' fields found", call. = FALSE)
-    }
-  }
+  cat_n("gbif_citation() for occ_search() and occ_data() is deprecated.")
+  cat_n("Use rgbif::occ_download() or rgbif::derived_dataset() instead.")
+}
```

Comment:
This goes beyond deprecated, and removes the function completely. I think that will need to wait until version 3.8.0, at least according to the rOpenSci developer guidelines: https://devguide.ropensci.org/evolution.html#functions-deprecate-defunct

Reply:
ok good point. I was not sure how to deprecate a method, but there probably is a way.